Fix minimal gui option to hide smoke behind biomass

keatsandyeats opened the pull request:
Activating the minimal gui option will now hide the smokey background
behind the alien biomass display. This behavior was broken by a
vanilla update c. April 2016.

keatsandyeats committed:
Fix minimal gui option to hide smoke behind biomass

Activating the minimal gui option will now hide the smokey background
behind the alien biomass display. This behavior was broken by a
vanilla update c. April 2016.

Mendasp commented:
If you're adding a new file for this you might want to remove the old code that handled this by doing a search for "biomass" in all the NS2Plus files, it seems the old code is both in GUIMinimap.lua and GUIAlienHUD.lua (one for players and one for the alien comm) - also if possible you should consider trying to not override it every frame and just do it after the original code toggles it (and on script reset), for performance.

Cheers.

keatsandyeats committed:
Refactor the fix for minimal gui hiding biomass smoke

Turn the smoke texture transparent instead of hiding it every
frame. The new code is now with the rest of the Alien GUI
changes.

keatsandyeats commented:
Thanks for the advice, Mendasp. I've now set the texture invisible instead of hiding it every frame. And instead of adding the new file, the changes are now with the other Alien GUI changes.

GhoulofGSG9 commented:
Okay I merged this PR into beta and will release ns2+ version 371 with the next ns2 update (317).
